rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -46,83 +44,52 @@ def required_components(cls) -> List[Type[Component]]:
-        "use_word_boundaries": True,
     }
 
     def __init__(
         self,
         component_config: Optional[Dict[Text, Any]] = None,
         known_patterns: Optional[List[Dict[Text, Text]]] = None,
-        pattern_vocabulary_stats: Optional[Dict[Text, int]] = None,
         finetune_mode: bool = False,
     ) -> None:
         """Constructs new features for regexes and lookup table using regex expressions.
 
         Args:
             component_config: Configuration for the component
             known_patterns: Regex Patterns the component should pre-load itself with.
-            pattern_vocabulary_stats: Statistics about number of pattern slots filled
-                and total number available.
             finetune_mode: Load component in finetune mode.
         """
         super().__init__(component_config)
 
         self.known_patterns = known_patterns if known_patterns else []
         self.case_sensitive = self.component_config["case_sensitive"]
-        self.number_additional_patterns = self.component_config[
-            "number_additional_patterns"
-        ]
         self.finetune_mode = finetune_mode
-        self.pattern_vocabulary_stats = pattern_vocabulary_stats
-
-        if self.finetune_mode and not self.pattern_vocabulary_stats:
-            # If the featurizer is instantiated in finetune mode,
-            # the vocabulary stats for it should be known.
-            raise rasa.shared.exceptions.InvalidParameterException(
-                f"{self.__class__.__name__} was instantiated with"
-                f" `finetune_mode=True` but `pattern_vocabulary_stats`"
-                f" was left to `None`. This is invalid since the featurizer"
-                f" needs vocabulary statistics to featurize in finetune mode."
-            )
-
-    @lazy_property
-    def vocabulary_stats(self) -> Dict[Text, int]:
-        """Computes total vocabulary size and how much of it is consumed.
-
-        Returns:
-            Computed vocabulary size and number of filled vocabulary slots.
-        """
-        if not self.finetune_mode:
-            max_number_patterns = (
-                len(self.known_patterns) + self._get_num_additional_slots()
-            )
-            return {
-                "pattern_slots_filled": len(self.known_patterns),
-                "max_number_patterns": max_number_patterns,
-            }
-        else:
-            self.pattern_vocabulary_stats["pattern_slots_filled"] = len(
-                self.known_patterns
+        if self.component_config["number_additional_patterns"]:
+            rasa.shared.utils.io.raise_deprecation_warning(
+                "The parameter `pattern_vocabulary_stats` has been deprecated "
+                "since the pipeline does not create an extra buffer for new vocabulary anymore. "
+                "Any value assigned to this parameter will be ignored. "
+                "You can omit specifying `pattern_vocabulary_stats` in future runs."
```

Comment:
```suggestion
                "The parameter `number_additional_patterns` has been deprecated "
                "since the pipeline does not create an extra buffer for new vocabulary anymore. "
                "Any value assigned to this parameter will be ignored. "
                "You can omit specifying `number_additional_patterns` in future runs."
```